src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java

```diff
@@ -49,27 +52,23 @@ public static PerformerNameAndAddress fromString(String edifactString) {
             );
         }
 
-        String[] keySplit = Split.byPlus(edifactString);
-        String[] colonSplit = Split.byColon(keySplit[2]);
-        String identifier = colonSplit[0];
+        final String[] keySplit = Split.byPlus(edifactString);
+        final String[] colonSplit = Split.byColon(keySplit[2]);
 
-        final boolean isOrganization = StringUtils.isBlank(identifier);
-        if (isOrganization) {
-            // if identifier is blank - organization/department
-            String partyName = keySplit[4];
-
-            return PerformerNameAndAddress.builder()
-                .partyName(partyName)
-                .build();
+        String identifier = null;
+        HealthcareRegistrationIdentificationCode code = null;
+        if (colonSplit.length > 1 && StringUtils.isNotBlank(colonSplit[0])) {
+            identifier = colonSplit[0];
+            code = StringUtils.isNotBlank(colonSplit[1])
+                ? HealthcareRegistrationIdentificationCode.fromCode(colonSplit[1]) : null;
         }
 
-        String code = colonSplit[1];
-        String practitionerName = keySplit[4];
+        final String name = keySplit.length > 3 && StringUtils.isNotBlank(keySplit[4]) ? keySplit[4] : null;
```

Comment:
#94 introduces `arrayGetSafe` which we could use here, with some tweaking if you want to keep the conditional logic about only reading a code if there's an identifier:
```
        var identifier = arrayGetSafe(colonSplit, 0).orElse(null);
        var code = arrayGetSafe(colonSplit, 1)
            .map(HealthcareRegistrationIdentificationCode::fromCode)
            .orElse(null);
        var name = arrayGetSafe(keySplit, 4).orElse(null);
```

Reply:
That's a nice idea, i'll take a look 😄

Reply:
I've updated the above to use the `arrayGetSafe` method